pkgs/development/compilers/ante/default.nix

```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, lib
+, libffi
+, libxml2
+, llvmPackages_13
+, ncurses
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "ante";
+  version = "unstable-2022-08-22";
+  src = fetchFromGitHub {
+    owner = "jfecher";
+    repo = "ante";
+    rev = "8b708d549c213c34e4ca62d31cf0dd25bfa7b548";
+    sha256 = "sha256-s8nDuG32lI4pBLsOzgfyUGpc7/r0j4EhzH54ErBK7A0=";
+  };
+  cargoSha256 = "sha256-29D7kPG7vop9lIxWQnaHkCTRY8YsCjERRCOvbU7oemQ=";
+
+  nativeBuildInputs = [ llvmPackages_13.llvm ];
```

Comment:
Isn't this included in rust already?

Reply:
`llvmPackages` are indeed included with rustc that's part of `rustPlatform`. I could in theory extract them with `rustPlatform.rust.rustc.passthru.llvmPackages`, however llvm-sys requires a [specific minor version of llvm](https://crates.io/crates/llvm-sys#llvm-compatibility), which for ante is 13.0.*, and the version of llvmPackages bundled with rustc is 14.0.1.

I'm generating the system variable pointing to the location of the llvmPackages using the actual version of that package, so I could either set the `LLVM_SYS_#{llvm-sys-ver}_IGNORE_BLOCKLIST` variable to true, rebuild the rust platform with the appropriate version of llvm and then build ante with it, or keep things as they are.

Though with the `LLVM_SYS_${llvm-sys-ver}_PREFIX` variable set, the build seems to succeed without the nativeBuildInputs, so they probably can be removed, but `llvmPackages_13` will probably need to still be used for setting that build variable.

Reply:
It does build without that package listed there, but I feel like it might be confusing to only include the reference to the package in an environment variable in the `preBuild` hook, and there might be more things depending on the specific version of `llvmPackages` in the future.